fe/fe-core/src/main/java/com/starrocks/sql/ast/UserVariable.java

```diff
@@ -97,55 +94,86 @@ public String toSql() {
     public void deriveUserVariableExpressionResult(ConnectContext ctx) {
         QueryStatement queryStatement = ((Subquery) unevaluatedExpression).getQueryStatement();
         ExecPlan execPlan = StatementPlanner.plan(queryStatement,
-                ConnectContext.get(), TResultSinkType.VARIABLE);
+                ConnectContext.get(), TResultSinkType.MYSQL_PROTOCAL);
```

Comment:
What's the different behaviors between `VARIABLE ` and `MYSQL_PROTOCAL `?

Reply:
VARIABLE type is the old implementation. It cast the output col to string and use the thrift protocal toSerialize/Deserilize. Now, we dont's support cast a array to string.  `MYSQL_PROTOCAL` is the normal way to encode the result to bytes and return to user client.

Reply:
> use mysql_protocal is unnecessary? it's will effect other type?
 I think mysql_protocal is more common, because it support all types. It's does't affect other types, becuase the client use the same way to decode the final value.

Reply:
for complex type, MYSQL_PROTOCAL write the value by string way in BE, it's hard to support nested type in feature? we need parse it manual in fe. how about HTTP_PROTOCAL ( json format) ? it's support all types now, and more easy handle it by gson read